plugins/vocabbuilder.koplugin/main.lua

```diff
@@ -550,8 +549,15 @@ function VocabItemWidget:initItemWidget()
     local right_side_width
     local right_widget
     if not self.show_parent.is_edit_mode and self.item.due_time <= os.time() then
+        local temp_button = Button:new{
+            text = _("Got it"),
+            padding_h = Size.padding.large
+        }
+        local review_button_width = temp_button:getSize().w
+        temp_button:setText(_("Forgot"))
+        review_button_width = math.min(math.max(review_button_width, temp_button:getSize().w), Screen:getWidth()/4)
+        temp_button:free()
```

Comment:
Dunno if this is done for each horizontal item - but if yes, to save a minor bit on resources, you could save it as self.review_button_width - and save the computed sizes as module properties.
Just wrap this in `if not self.review_button_width then ... `

Reply:
I ended up putting the variable in the file scope as a local variable. Tried declaring it in the :new{} (initializer?) and declare it on the fly, both still calculate multiple times. I haven't found how to do a static property in Lua :/

Reply:
You don't really need to do it only once for the lifetime of a koreader session.
Doing it once after the plugin is instantiated (so, to be redone once after each book opening) , and when needed so the code stays in context, is just fine. Many modules do it that way:
```lua
if not self.review_button_width then 
  self.review_button_width = compute it
end
-- and just use self.review_button_width 
```

Reply:
I tried it and had a debug message printed each time it's called, and it prints as many times as the number of words shown. Or I haven't found the right place to insert it? Since what changes the width is the device dpi and translations, both of which require a restart, having it only computed once seems alright?

Reply:
You're sure you have added `self.` in front of all your previously-local `review_button_width` ?

Reply:
Oh, ok, you are in VocabItemWidget - so `self` is each time a new instance - so it won't work as the holder :)
So, yes, file scope local variable is the solution - sorry :)